src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +379,168 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    public void validateSummaryTag(DetailNode ast) {
+        if (getInlineSummary(ast).isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final String inlineSummaryDoc = getInlineSummary(ast);
+            if (isPeriodAtEnd(inlineSummaryDoc, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+            else if (isSummaryValid(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
```

Comment:
why not simple `period.equals(summarySentence[summarySentence.length() - 1])` ?

Reply:
not necessarily period will be of index 1, as it can be set be user. We can use substring method to compare string but that also needs to be checked if period if string length 0 or not, also that is bit complex to understand.